ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java

```diff
@@ -144,8 +148,19 @@ private static boolean isTransformedTouchPointInView(
       localX = localXY[0];
       localY = localXY[1];
     }
-    if ((localX >= 0 && localX < (child.getRight() - child.getLeft()))
-        && (localY >= 0 && localY < (child.getBottom() - child.getTop()))) {
+    float hitSlopTop = 0;
```

Comment:
I think it will be much simpler to understand the hit-testing rules if hitSlop affects only the size of the hit box and not the priority. Later we can add a new property like zIndex or hitPriority that affects the priority.